Cauli/Configuration/Configuration.swift

```diff
@@ -57,12 +66,17 @@ public struct Configuration {
 
     /// The `storageCapacity` defines the capacity of the storage.
     public let storageCapacity: StorageCapacity
+    
+    /// Pass in a callback that will be executed on each `Record` before it is persisted to a `Storage`.
+    /// This allows you to modify requests and responses after they are executed but before they are passed along to other florets.
+    public var prePersistHook: PrePersistHook?
```

Comment:
For the property name here I will just throw in a few thoughts and ideas I have:

* `willStoreRecord`: this ties into the functions in the Floret protocol 
* `preStorageRecordModifier`: this keeps my proposed `RecordModifier` and replaces persist with storage to keep the similarity here.

Reply:
I commited the second one for now. I also like `willStoreRecord`, but it doesn't tell the user that explicitly that you can modify the `Record` here. That may not be necessary though, but we still can change it.

Reply:
Nice! I am fine with either.